### PR TITLE
fix: resolve issues with start date on handler's side (HL-1270)

### PR DESCRIPTION
**Pull request by sirtawast**
## Description :sparkles:

* Refactor and change some constants regarding start date validation
* Add notification on handler application edit when changing start / end date too far from recommended range
* Fix a nasty bug that crashed the whole app

image

### CODE REVIEW

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://helsinkilisa-pr2948.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**APPLICANT branch is deployed to platta: https://helsinkilisa-ui-pr2948.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://helsinkilisa-ui-handler-pr2948.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-pr2948.dev.hel.ninja :laughing::tada::tada::tada:

File: frontend/benefit/handler/src/components/applicationForm/formContent/useFormContent.ts [hunk 1]
```diff
@@ -118,6 +139,7 @@ const useFormContent = (
     clearDatesValues,
     clearAlternativeAddressValues,
     getErrorMessage,
+    displayPastApplicationDatesWarning,
```
Comment (EmiliaMakelaVincit) [inline frontend/benefit/handler/src/components/applicationForm/formContent/useFormContent.ts:142]: I guess it's a bit unconventional to output this as a function when it doesn't take any parameters and its return value is already constant within any specific render cycle.

**Review by EmiliaMakelaVincit — APPROVED** [review]
One note that may be acted on, but seems fine.

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://helsinkilisa-ui-handler-pr2948.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://helsinkilisa-pr2948.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-handler-pr2948.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-pr2948.dev.hel.ninja :laughing::tada::tada::tada:

**Pull request merged**